daemon/policy.go

```diff
@@ -258,6 +258,14 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (newRev uint
 		}
 	}
 	newRev = d.policy.AddListLocked(rules)
+
+	// The rules are added, we can begin ToFQDN DNS polling for them
+	// Note: api.FQDNSelector.sanitize checks that the matchName entries are
+	// valid. This error should never happen (of course).
+	if err := d.dnsRuleGen.StartManageDNSName(rules); err != nil {
```

Comment:
Any concerns here in terms of resource contention about holding the policy repository lock while the `dnsRuleGen` lock is held as well?

Reply:
Generally no, `StartManageDNSName` should only do bookkeeping inside RuleGen, so only hold that lock. Calling back into PolicyAdd would/should only happen on a DNS update. This can happen while this specific invocation happens (and the lock is held) but that is a different thread so it won't deadlock.